tests/unittest/_torch/test_fp8_block_scale_gemm.py

```diff
@@ -181,144 +147,45 @@ def test_fp8_blockscale_gemm_trtllmgen(dtype):
-                      torch.empty((num_batches, m, k // 128),
-                                  device='cuda',
-                                  dtype=torch.float))
-    y_fp8, y_scale = (torch.empty_like(y, dtype=torch.float8_e4m3fn),
-                      torch.empty((num_batches, (n + 127) // 128, k // 128),
-                                  device='cuda',
-                                  dtype=torch.float))
-
-    for i in range(num_batches):
-        x_fp8[i], x_scale_i = torch.ops.trtllm.fp8_quantize_1x128(x[i])
-        x_scale[i] = x_scale_i.view(x_scale[i].shape)
-        y_fp8[i], y_scale[i] = per_block_cast_to_fp8(y[i])
-
-    return x_fp8, x_scale, y_fp8, y_scale, ref_out
-
-
-@pytest.mark.skipif(
-    getSMVersion() != 90,
-    reason="Op only supported on Hopper",
-)
-@pytest.mark.parametrize("ms", [[256, 256], [128, 64, 64], [16, 24, 48]])
-@pytest.mark.parametrize("k, n", [(7168, 4096), (2048, 7168)])
-def test_fp8_block_scaling_moe_gemm(ms, k, n):
-    offset_cpu = [0] + list(itertools.accumulate(ms))
-    offset = torch.tensor(offset_cpu, device='cuda', dtype=torch.int64)
-    x_fp8, x_scale, y_fp8, y_scale, ref_out = construct_grouped(ms, k, n)
-    x_fp8, x_scale = change_to_offset_layout(ms, x_fp8, x_scale)
-    out = torch.ops.trtllm.fp8_block_scaling_moe_gemm(x_fp8, y_fp8, x_scale,
-                                                      y_scale, offset)
-    diff = calc_diff(out, ref_out)
-    assert diff < 1e-3
-    torch.testing.assert_close(out, ref_out, atol=1e-3, rtol=1e-3)
-
-
-@pytest.mark.skipif(
-    getSMVersion() != 90,
-    reason="Op only supported on Hopper",
-)
-@pytest.mark.parametrize("batch_size, m", [(1, 1024), (2, 512), (4, 256)])
-@pytest.mark.parametrize("k, n", [(7168, 4096), (2048, 7168)])
-def test_fp8_block_scaling_bmm(batch_size, m, k, n):
-    torch.random.manual_seed(0)
-    x_fp8, x_scale, y_fp8, y_scale, ref_out = construct_batched(
-        batch_size, m, k, n)
-    output = torch.ops.trtllm.fp8_block_scaling_bmm(x_fp8, y_fp8, x_scale,
-                                                    y_scale)
-    diff = calc_diff(output, ref_out)
-    assert diff < 1e-3
-    torch.testing.assert_close(output, ref_out, atol=1e-3, rtol=1e-3)
+def run_test_in_subprocess(env, test_file):
+    # Create a copy of the current environment
+    process_env = os.environ.copy()
+
+    # Update with the new environment variables
+    process_env.update(env)
+
+    # Run the test in a subprocess
+    result = subprocess.run([sys.executable, '-m', 'pytest', test_file, '-v'],
+                            capture_output=True,
+                            text=True,
+                            env=process_env)
+
+    # Print the output
+    print(result.stdout)
+    if result.stderr:
+        print(result.stderr)
+
+    # Return the exit code
+    return result.returncode
+
+
+@pytest.mark.parametrize("env", [
+    {
+        'TRTLLM_DG_ENABLED': '0'
```

Comment:
If we use function to query these env, not static vars. we may dont need to have another subprocess to launch pytest

Reply:
Problem here is that pytest runs tests in parallel so the env vars set later will overwrite those set previously.